### PR TITLE
added login form and logout button on the homepage component

### PR DESCRIPTION
**Pull request by DmitryApan**
image
image
image

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -139,8 +148,7 @@
 .avatar-lay {
   position: absolute;
   right: 50%;
-  /* right: -11px; */
-  
+  /* right: -11px; */
```
Comment (js-cowboy) [inline src/App.css]: do you need it?

File: src/App.css [hunk 2]
```diff
@@ -170,9 +179,81 @@
...
+
+  font-size: 15px;
+  font-weight: bold;  
+}
+
+.login-submit-button {  
+  width: 100px;
+  border: 0px;
+  background-color: rgb(100, 145, 182);
+
```
Comment (js-cowboy) [inline src/App.css]: do you need space between lines?

File: src/App.css [hunk 3]
```diff
@@ -170,9 +179,81 @@
...
+}
+
+.login-link {
+  margin-left: 80px;
+  text-decoration: none;
+  font-weight: bold;
+  color: rgb(63, 97, 124);
+}
+
+input::-webkit-input-placeholder {
```
Comment (js-cowboy) [inline src/App.css]: I guess you can use `::placeholder` without any prefix

File: src/App.css [hunk 4]
```diff
@@ -170,9 +179,81 @@
...
+  color: rgb(224, 222, 222);
+ } 
+
+input::-moz-placeholder {
```
Comment (js-cowboy) [inline src/App.css]: the same

File: src/App.js [hunk 5]
```diff
@@ -63,6 +72,31 @@ class App extends React.Component {
...
+                    </Route>
+                    <Route path='/login'>
+                        <LoginPage {
+                            ...{
+                                buttonText: 'Login',
+                                linkText: 'Logout?',
+                                linkUrl: '/register',
+                                onSubmitLoginForm: this.handlerSubmitLoginForm,
+                                onChangeLoginInput: this.handlerChangeLoginInput
+                            }
```
Comment (js-cowboy) [inline src/App.js]: looks odd, why do you need spread here? just use props

File: src/App.js [hunk 6]
```diff
@@ -63,6 +72,31 @@ class App extends React.Component {
...
+                        } />
+                    </Route>
+                    <Route path='/register'>
+                    <LoginPage {
+                            ...{
+                                buttonText: 'Logout',
+                                linkText: 'Login?',
+                                linkUrl: '/login',
+                                onSubmitLoginForm: this.handlerSubmitLoginForm,
+                                onChangeLoginInput: this.handlerChangeLoginInput
```
Comment (js-cowboy) [inline src/App.js]: the same

File: src/App.js [hunk 7]
```diff
@@ -63,6 +72,31 @@ class App extends React.Component {
...
+                                buttonText: 'Login',
+                                linkText: 'Logout?',
+                                linkUrl: '/register',
+                                onSubmitLoginForm: this.handlerSubmitLoginForm,
+                                onChangeLoginInput: this.handlerChangeLoginInput
+                            }
+                        } />
+                    </Route>
+                    <Route path='/register'>
+                    <LoginPage {
```
Comment (js-cowboy) [inline src/App.js]: missed tab here

File: src/HomePage.js [hunk 8]
```diff
@@ -80,14 +81,19 @@ export class HomePage extends React.Component {
...
+                    {statuses && statuses.map(status => ( 
+                        <Section 
+                            status={status} 
+                            cards={dataByStatuses[status] || []}                        
+                            onDeleteCard={this.handleDeleteCard}
+                            onModalInfo={this.handleModalInfo}
+                        />))}
+                </div>
+                <div class="homepage-region-logout flex-column">
+                    <Link class="button-logout" to="/logout" >Logout</Link>
```
Comment (js-cowboy) [inline src/HomePage.js]: you don't need space before closing tag

File: src/LoginPage.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+import {Link} from 'react-router-dom';
+
+export function LoginPage({buttonText, linkText, linkUrl, onSubmitLoginForm, onChangeLoginInput}) {
+    return(
```
Comment (js-cowboy) [inline src/LoginPage.js]: you need space before brace

File: src/LoginPage.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+        <div class="login-overlay"> 
+            <div class="flex-center login-window" >
+                <form onSubmit={onSubmitLoginForm} class="flex-column">
+                    <input onChange={onChangeLoginInput} class="login-textbox-email" type="text" name="userEmail" placeholder="Email or phone number" />
+                    <input onChange={onChangeLoginInput} class="login-textbox-password" type="password" name="userPassword" placeholder="Password" />
+                    <div class="login-region-submit">
+                        <input class="login-submit-button" type='submit' name={buttonText} value={buttonText} />
```
Comment (js-cowboy) [inline src/LoginPage.js]: double quotes for html, single for js

Comment (js-cowboy) [reply src/LoginPage.js]: btw you could use button instead of input type submit

File: src/App.css (hunk 1 above)
Comment (DmitryApan) [reply src/App.css]: > do you need it?

No

File: src/App.css (hunk 2 above)
Comment (DmitryApan) [reply src/App.css]: > do you need space between lines?

not sure i need them.
I will remove all the rusty lines in the styles

File: src/App.css (hunk 3 above)
Comment (DmitryApan) [reply src/App.css]: Check it out

File: src/App.js (hunk 5 above)
Comment (DmitryApan) [reply src/App.js]: > looks odd, why do you need spread here? just use props

should I create fields in the status of the App component: email and password?

File: src/App.js [hunk 11]
```diff
@@ -74,29 +74,24 @@ class App extends React.Component {
...
-                                linkText: 'Logout?',
-                                linkUrl: '/register',
-                                onSubmitLoginForm: this.handlerSubmitLoginForm,
-                                onChangeLoginInput: this.handlerChangeLoginInput
-                            }
-                        } />
+                        <LoginPage                                                        
+                            buttonText={'Login'}
+                            linkText={'Logout?'} 
+                            linkUrl={'/register'}
```
Comment (js-cowboy) [inline src/App.js]: it's easier to pass double quoted strings but this way also works, it's up to you

File: src/LoginPage.js [hunk 12]
```diff
@@ -3,14 +3,14 @@ import React from 'react';
...
-    return(
+    return (
         <div class="login-overlay"> 
             <div class="flex-center login-window" >
                 <form onSubmit={onSubmitLoginForm} class="flex-column">
                     <input onChange={onChangeLoginInput} class="login-textbox-email" type="text" name="userEmail" placeholder="Email or phone number" />
                     <input onChange={onChangeLoginInput} class="login-textbox-password" type="password" name="userPassword" placeholder="Password" />
                     <div class="login-region-submit">
-                        <input class="login-submit-button" type='submit' name={buttonText} value={buttonText} />
+                        <button class="login-submit-button" type="submit" name={buttonText}>{buttonText}</button>
```
Comment (js-cowboy) [inline src/LoginPage.js]: type submit is redundant here

File: src/App.js [hunk 13]
```diff
@@ -74,29 +74,24 @@ class App extends React.Component {
...
+                            buttonText={'Logout'}
+                            linkText={'Login'}
+                            linkUrl={'/login'}
+                            onSubmitLoginForm={this.handlerSubmitLoginForm}
+                            onChangeLoginInput={this.handlerChangeLoginInput}                            
+                        />
                     </Route>
-                    <Route path='/logout'>
-                        
+                    <Route path='/logout'>
```
Comment (js-cowboy) [inline src/App.js]: double quotes for html

File: src/App.js [hunk 14]
```diff
@@ -70,30 +70,30 @@ class App extends React.Component {
...
                     </Route>
-                    <Route path='/login'>
+                    <Route path="/login">
                         <LoginPage                                                        
-                            buttonText={'Login'}
-                            linkText={'Logout?'} 
-                            linkUrl={'/register'}
+                            buttonText={"Login"}
+                            linkText={"Logout?"} 
+                            linkUrl={"/register"}
```
Comment (js-cowboy) [inline src/App.js]: you don't need curly braces here, since it's just string

File: src/App.js [hunk 15]
```diff
@@ -70,30 +70,30 @@ class App extends React.Component {
...
                     </Route>
-                    <Route path='/register'>
+                    <Route path="/register">
                         <LoginPage                            
-                            buttonText={'Logout'}
-                            linkText={'Login'}
-                            linkUrl={'/login'}
+                            buttonText={"Logout"}
+                            linkText={"Login"}
+                            linkUrl={"/login"}
```
Comment (js-cowboy) [inline src/App.js]: same

File: src/App.css [hunk 16]
```diff
@@ -178,9 +165,78 @@
...
+  width: 300px; 
+  border: 1px solid rgb(206, 205, 205);  
+  box-sizing: border-box;
+  font-size: 15px;
+  font-weight: bold;  
+}
+
+.login-submit-button {  
+  width: 100px;
+  border: 0px;
```
Comment (js-cowboy) [inline src/App.css]: you can use border: none when you don't need border

File: src/App.js [hunk 17]
```diff
@@ -19,53 +21,117 @@ class App extends React.Component {
...
+            isLogin: !!email,
+            dataCard: email ? await this.dataRequest() : {}                               
+        });        
+    }   
+    
+    async dataRequest() {
+        let dataCard = {
+            statuses: [],
+            dataByStatuses: {},
+            email: localStorage.getItem('email')
```
Comment (js-cowboy) [inline src/App.js]: isn't it duplication of 30 and 33 line from above?

File: src/App.js [hunk 18]
```diff
@@ -19,53 +21,117 @@ class App extends React.Component {
...
-        });        
-    }   
-    
     updateData = (value) => {
         this.setState(value);
     }    
 
+    handlerChangeLoginInput = (event) => {
+        const value = event.target.value;
+        const name = event.target.name;
```
Comment (js-cowboy) [inline src/App.js]: destructure event or even target there

File: src/App.js [hunk 19]
```diff
@@ -19,53 +21,117 @@ class App extends React.Component {
...
             <Router>
-                <Switch>                             
-                    <Route exact path='/'>
-                        <HomePage {...state} updateData={updateData} />                                  
-                    </Route>
-                    <Route path='/:id' >
-                        <CardPage {...this.state.dataCard} />
-                    </Route>                                                        
+                <Switch>
+                    {('isLogin' in state) && (isLogin
```
Comment (js-cowboy) [inline src/App.js]: pretty weird check, why do you need it?

Comment (js-cowboy) [reply src/App.js]: @DmitryApan you missed it

**Review by js-cowboy — APPROVED** [review]

**Pull request merged**